### PR TITLE
shipit_uplift: Retrieve modified files directly with pushes and avoid raw-rev requests

### LINKED ISSUE #816: shipit_uplift: Retrieve pushes with modified files so we can avoid the raw-rev request in the coverage by changeset endpoint

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #816 

This is also increasing the size of some caches.

### CODE REVIEW

**Comment by marco-c** [comment]
There's something wrong in the requirements.nix generated by `./please tools update-dependencies shipit-uplift`. @La0 would you mind running it for me and sending me the requirements.nix and requirements_frozen.txt files?

File: src/shipit_uplift/shipit_uplift/coverage.py [hunk 1]
```diff
@@ -6,15 +6,22 @@
...
 import json
 from functools import lru_cache
+from cachetools import LRUCache
 import requests
 
+from shipit_uplift import secrets
 
-@lru_cache(maxsize=128)
+
+@lru_cache(maxsize=2048)
```
Comment (La0) [inline src/shipit_uplift/shipit_uplift/coverage.py]: Why the increase ?

File: src/shipit_uplift/shipit_uplift/coverage.py [hunk 2]
```diff
@@ -200,48 +210,85 @@ def get_latest_build():
...
+
+    for pushid, pushdata in data['pushes'].items():
+        pushid = int(pushid)
+
+        push_to_changeset_cache[pushid] = pushdata['changesets'][-1]['node']
+
+        for changeset in pushdata['changesets']:
+            is_merge = any(text in changeset['desc'] for text in ['r=merge', 'a=merge'])
+
+            if not is_merge:
```
Comment (La0) [inline src/shipit_uplift/shipit_uplift/coverage.py]: Nit: i would write first the "positive" case, then the other one. Feel free to ignore.

File: src/shipit_uplift/shipit_uplift/worker.py [hunk 3]
```diff
@@ -5,26 +5,12 @@
...
-    required=[],
-    existing={x: os.environ.get(x) for x in ['REDIS_URL'] if x in os.environ},
-    taskcluster_client_id=os.environ.get('TASKCLUSTER_CLIENT_ID'),
-    taskcluster_access_token=os.environ.get('TASKCLUSTER_ACCESS_TOKEN'),
-)
-
-REDIS_URL = secrets['REDIS_URL'] if 'REDIS_URL' in secrets else 'redis://localhost:6379'
-
-conn = redis.from_url(REDIS_URL)
+conn = redis.from_url(secrets.REDIS_URL)
```
Comment (La0) [inline src/shipit_uplift/shipit_uplift/worker.py]: 👍

File: src/shipit_uplift/shipit_uplift/coverage.py [hunk 4]
```diff
@@ -94,12 +101,16 @@ class CodecovCoverage(Coverage):
     URL = 'https://codecov.io/api/gh/marco-c/gecko-dev'
 
     @staticmethod
-    @lru_cache(maxsize=32)
+    def _get(url):
+        return requests.get(CodecovCoverage.URL + url + '?access_token=%s' % secrets.CODECOV_TOKEN)
```
Comment (La0) [inline src/shipit_uplift/shipit_uplift/coverage.py]: I would rather use a `.format` here to increase readability.

File: src/shipit_uplift/shipit_uplift/coverage.py (hunk 1 above)
Comment (marco-c) [reply src/shipit_uplift/shipit_uplift/coverage.py]: The larger cache should speed up more requests.

File: src/shipit_uplift/shipit_uplift/coverage.py (hunk 4 above)
Comment (marco-c) [reply src/shipit_uplift/shipit_uplift/coverage.py]: Done in the latest commit.

**Pull request merged**